### DISCUSSION TITLE
Support for German fiscal elements (TSE) from EPSON

### DISCUSSION THREAD

**Discussion by strubbi77** [python-escpos/python-escpos#490] [Ideas]
I'm starting to implement a basic support for German fiscal elements (TSE) from EPSON in python using the xml-Protocol from EPSON. 
Could this also be part of python-escpos or is it out of scope of this project? 

Best regards
Thomas

**Comment by patkan** [comment]
Hi Thomas, this is very likely out of scope, but nevertheless, did you find anything interesting?

**Reply by strubbi77** [reply]
I had to buy additionally the Hub from Epson (https://www.epson.de/de_DE/produkte/retail/pos-drucker/fiscal-devices/fiscal-server-for-germany-%28eps-tse-server-3%29/p/29705) and got it to work with https://pypi.org/project/tse.py/

It might be possible to do it directly, but the efford would have been to high for me.

**Reply by patkan** [reply]
Thank you for the feedback!
